File: ObserverPattern/python/Audience.py
import abc

class Audience(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    def update(self):
        return NotImplemented
    
    
class TwitchAudience(Audience):
    def __init__(self, accountID, nickname) -> None:
        super().__init__()
        self.accountID= accountID
        self.nickname = nickname

    def update(self, streamInfo):
        print(f'\tAudience {self.nickname} receive the message: {streamInfo}')

File: ObserverPattern/cpp/src/audience.cpp
#include <string>
#include <iostream>
#include "header/audience.h"

using namespace std;

TwitchAudience::TwitchAudience(int accountID, string nickname){
    this->accountID = accountID;
    this->nickname = nickname;
}

void TwitchAudience::update(string streamInfo){
    cout << "\tAudience " << nickname << " receive the message: " << streamInfo << endl;
}